lte/gateway/c/core/oai/tasks/s1ap/s1ap_state.cpp

```diff
@@ -154,62 +154,56 @@ void delete_s1ap_ue_state(imsi64_t imsi64) {
   S1apStateManager::getInstance().clear_ue_state_db(imsi_str);
 }
 
-bool get_mme_ue_ids_no_imsi(uint32_t keyP, uint64_t const dataP, void* argP,
-                            void** resultP) {
-  hash_key_t** mme_id_list = (hash_key_t**)resultP;
-  uint32_t* num_ues_checked = (uint32_t*)argP;
+void get_mme_ue_ids_no_imsi(uint32_t keyP, uint64_t const dataP,
+                            uint32_t* num_ues_checked,
+                            std::vector<uint32_t>* mme_id_list,
+                            hash_table_ts_t* s1ap_ue_state) {
```

Comment:
1. Since this function operates on one entry, name it as a checker
2. Rename the parameters for better readability

```suggestion
bool check_imsi_for_mme_ue_id(uint32_t mme_ue_id, uint64_t const comp_s1ap_id,
                            hash_table_ts_t* s1ap_ue_state) {
```